Classes/common/CDTIncrementalStore.m

```diff
@@ -2208,6 +2221,75 @@ - (id)executeSaveRequest:(NSSaveChangesRequest *)saveRequest
     return @[];
 }
 
+- (id)executeBatchUpdateRequest:(NSBatchUpdateRequest *)updateRequest
+                    withContext:(NSManagedObjectContext *)context
+                          error:(NSError **)error
+{
+    NSError *err = nil;
+
+    NSFetchRequest *fetchRequest = [[NSFetchRequest alloc] init];
+    fetchRequest.entity = updateRequest.entity;
+    fetchRequest.predicate = updateRequest.predicate;
+
+    NSDictionary *query = [self queryForFetchRequest:fetchRequest error:&err];
+
+    if (!query) {
+        if (error) *error = err;
+        return nil;
+    }
+
+    err = nil;
+    CDTQueryResult *hits = [self.indexManager queryWithDictionary:query options:nil error:&err];
+    // hits == nil is valid, get rid of this once tested
+    if (!hits && err) {
+        if (error) *error = err;
+        return nil;
+    }
+
+    // Note: This dictionary is *not* the same dictionary specified in the original request.
+    // The keys have all been transformed from attribute names into NSAttributeDescriptions
+    NSDictionary *updates = updateRequest.propertiesToUpdate;
+
+    NSEntityDescription *entity = [updateRequest entity];
+
+    NSMutableArray *results = [NSMutableArray array];
+    for (CDTDocumentRevision *rev in hits) {
+        NSManagedObjectID *moid = [self newObjectIDForEntity:entity referenceObject:rev.docId];
+        NSManagedObject *mo = [context objectWithID:moid];
+
+        for (NSAttributeDescription *attr in [updates allKeys]) {
+            id newValue = [updates[attr] constantValue];
+            [mo setValue:newValue forKey:attr.name];
+        }
+
+        if (![self updateManagedObject:mo error:&err]) {
+            if (error) *error = err;
+            return nil;
+        }
+
+        [results addObject:moid];
+    }
+
+    CDTISBatchUpdateResult *updateResult = [[CDTISBatchUpdateResult alloc] init];
+    updateResult.resultType = updateRequest.resultType;
+
+    switch (updateRequest.resultType) {
+        case NSUpdatedObjectIDsResultType:  // Return the object IDs of the rows that were updated
+            updateResult.result = results;
+            break;
+
+        case NSUpdatedObjectsCountResultType:  // Return the number of rows that were updated
+            updateResult.result = @([results count]);
+            break;
+
+        case NSStatusOnlyResultType:  // Don't return anything
```

Comment:
Actually, we return an empty object.
In fact, perhaps we should just `updateResult.result = @(YES)`?